compiler_opt/rl/generate_bc_trajectories.py

```diff
@@ -102,3 +191,151 @@ def get_advice(self, state: time_step.TimeStep) -> np.ndarray:
           break
     self.curr_step += 1
     return policy_action
+
+
+class ExplorationWorker(worker.Worker):
+  """Class which implements the exploration for the given module.
+
+  Attributes:
+    loaded_module_spec: the module to be compiled and explored
+    use_greedy: indicates if the default/greedy policy is used to compile the
+      module
+    env: MLGO environment.
+    exploration_frac: how often to explore in a trajectory
+    max_exploration_steps: maximum number of exploration steps
+    exploration_policy_distr: distribution function from exploration policy.
+    reward_key: which reward binary to use, must be specified as part of
+      additional task args (kwargs).
+  """
+
+  def __init__(
+      self,
+      loaded_module_spec: corpus.LoadedModuleSpec,
+      clang_path: str,
+      mlgo_task: Type[env.MLGOTask],
+      use_greedy: bool = False,
+      exploration_frac: float = 1.0,
+      max_exploration_steps: int = 10,
+      exploration_policy_distr=None,
+      obs_action_specs: Optional[Tuple[time_step.TimeStep,
+                                       tensor_spec.BoundedTensorSpec,]] = None,
+      **kwargs,
+  ):
+    self.loaded_module_spec = loaded_module_spec
+    self.use_greedy = use_greedy
+    if not obs_action_specs:
+      obs_spec = None
+      action_spec = None
+    else:
+      obs_spec = obs_action_specs[0].observation
+      action_spec = obs_action_specs[1]
+
+    if 'reward_key' not in kwargs:
```

Comment:
why not make `reward_key` an explicit parameter? if you type it appropriately (`str`, not `Optional[str]`), we can get a type checking error if the user doesn't pass it.

Reply:
done.